pkg/cmd/extension/push/cmd.go

```diff
@@ -0,0 +1,132 @@
+type cmdOpts struct {
+	// toolchain to use to build the *.wasm file.
+	toolchain common.ToolchainOpts
+	// extension to use to specify the built *.wasm file.
+	extension runtime.ExtensionOpts
+	// pusher to use to specify options for pusher
+	pusher wasmimage.PusherOpts
+}
+
+func newCmdOpts() *cmdOpts {
+	return &cmdOpts{
+		toolchain: common.ToolchainOpts{
+			Name: toolchain.Default,
+		},
+		extension: runtime.ExtensionOpts{},
+		pusher:    wasmimage.NewPusherOpts(),
+	}
+}
+
+func (opts *cmdOpts) GetToolchainName() string {
+	return opts.toolchain.Name
+}
+
+func (opts *cmdOpts) ApplyTo(interface{}) {}
+
+func (opts *cmdOpts) Validate() error {
+	if err := opts.toolchain.Validate(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// NewCmd returns a command that pushes the built extension.
+func NewCmd() *cobra.Command {
+	opts := newCmdOpts()
+	cmd := &cobra.Command{
+		Use:   "push <image-reference>",
+		Short: "Push the built WASM extension to the OCI-compliant registry.",
+		Long: `
+Push the built WASM extension to the OCI-compliant registry. This command requires to login the target container registry with docker CLI`,
+		Example: `
+  # Push built WASM extension to the local docker registry.
+  getenvoy extension push localhost:5000/test/image-name:tag`,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+
+			if len(args) == 0 {
+				return errors.New("missing image-reference parameter")
+			}
+
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			imagePath := opts.extension.WasmFile
+			if imagePath == "" {
+				ws, err := workspaces.GetCurrentWorkspace()
+				if err != nil {
+					return err
+				}
+				tc, err := common.LoadToolchain(ws, opts)
+				if err != nil {
+					return err
+				}
+				imagePath = tc.GetBuildOutputWasmFile()
+			}
+			imageRef := args[0]
+			ref, err := reference.ParseNormalizedNamed(imageRef)
+			if err != nil {
+				return fmt.Errorf("invalid image-reference: %w", err)
+			}
+			ref = reference.TagNameOnly(ref)
+			if tagged, ok := ref.(reference.Tagged); ok {
+				cmd.Printf("Using default tag: %s\n", tagged.Tag())
+			}
+			pusher, err := wasmimage.NewPusher(opts.pusher.AllowInsecure, opts.pusher.UseHTTP)
+			if err != nil {
+				return fmt.Errorf("failed to push the wasm image: %w", err)
+			}
+			manifest, size, err := pusher.Push(imagePath, imageRef)
+			if err != nil {
+				return fmt.Errorf("failed to push the wasm image: %w", err)
+			}
+
+			cmd.Printf("Pushed %s\n", ref)
+			cmd.Printf("digest: %s size: %d\n", manifest.Digest, size)
+
+			return nil
+		},
+	}
+	cmd.PersistentFlags().StringVar(&opts.toolchain.Name, "toolchain", opts.toolchain.Name,
+		`Name of the toolchain to use, e.g. "default" toolchain that is backed by a Docker build container`)
+	cmd.PersistentFlags().BoolVar(&opts.pusher.AllowInsecure, "allow-insecure", opts.pusher.AllowInsecure, `Allow insecure registry`)
```

Comment:
I think we should describe what "insecure" means here. (e.g. allow insecure tls connection)